test/boostTest.cpp

```diff
@@ -21,4 +21,8 @@
  */
 
 #define BOOST_TEST_MODULE EthereumTests
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
```

Comment:
if I remember well in all compilers you can simply put a `(void)paramName;` somewhere in the function body in order to totally silence these warnings. Plus side? Should work on all compilers.